File: 矩阵/48.旋转图像.py
from collections import deque


class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        # 按照一定顺序将数字到队列中，再通过想要的顺序进行赋值
        queue = deque()
        d = len(matrix)  # 注意获取的是长度
        for i in range(d):
            for j in range(d):
                queue.append(matrix[i][j])

        for i in range(d - 1, -1, -1):  # range只会遍历到第二个参数的前一个数
            for j in range(d):
                matrix[j][i] = queue.popleft()

File: 矩阵/48.旋转图像.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var rotate = function (matrix) {
  const n = matrix.length;
  const queue = [];
  // 将元素按行顺序存入队列
  for (let i = 0; i < n; i++) {
    for (let j = 0; j < n; j++) {
      queue.push(matrix[i][j]);
    }
  }
  // 从最后一列到第一列
  for (let col = n - 1; col >= 0; col--) {
    for (let row = 0; row < n; row++) {
      matrix[row][col] = queue.shift();
    }
  }
};
